src/components/dropdown-menu/DropdownMenuItemBase.tsx

```diff
@@ -0,0 +1,183 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import styled from '@emotion/styled';
+import * as radixDropdownMenu from '@radix-ui/react-dropdown-menu';
+import { MouseEventHandler, ReactNode } from 'react';
+import { IconCheck } from '../icons/IconCheck';
+
+type CheckProps =
+  | {
+      isCheckable: true;
+      isChecked?: boolean;
+    }
+  | {
+      isCheckable?: false;
+      isChecked?: never;
+    };
+
+export type DropdownMenuItemBaseProps = CheckProps & {
+  ariaLabel?: string;
+  children: ReactNode;
+  helpText?: JSX.Element | string;
+  isDisabled?: boolean;
+  isDestructive?: boolean;
+  onClick?: MouseEventHandler<HTMLDivElement>;
+  prefix?: JSX.Element;
+  suffix?: JSX.Element;
+};
+
+export function DropdownMenuItemBase({
+  ariaLabel,
+  children,
+  helpText,
+  isCheckable,
+  isChecked,
+  isDisabled,
+  isDestructive,
+  onClick,
+  prefix,
+  suffix,
+}: Readonly<DropdownMenuItemBaseProps>) {
+  let checkMarkOrPlaceHolder = null;
+
+  if (isCheckable) {
+    checkMarkOrPlaceHolder = isChecked ? (
+      <IconCheck color="echoes-color-icon-selected" />
+    ) : (
+      <CheckmarkPlaceholder />
+    );
+  }
+
+  return (
+    <StyledDropdownMenuItemBase
+      disabled={isDisabled}
+      isCheckable={isCheckable}
+      isChecked={isChecked}
+      isDestructive={isDestructive}
+      onClick={isDisabled ? undefined : onClick}>
+      <StyledLeftHandSide>
+        {checkMarkOrPlaceHolder}
+
+        {prefix}
+
+        <StyledLabelAndHelpText aria-label={ariaLabel}>
```

Comment:
The aria label should be on the `StyledDropdownMenuItemBase` if we have one, that's the interactive part. But I wonder if it's really needed?